app/src/androidTest/kotlin/com/wire/android/core/crypto/CryptoBoxClientIntegrationTest.kt

```diff
@@ -60,11 +60,30 @@ class CryptoBoxClientIntegrationTest : InstrumentationTest() {
         bobClient.delete()
     }
 
+    @Test
+    fun givenAliceSessionDoesNotExistsForBob_whenCheckingExistence_itShouldReturnFalse() {
+        bobClient.doesSessionExists(aliceSessionId)
+            .shouldSucceed {
+                it shouldBeEqualTo false
+        }
+    }
+
+    @Test
+    fun givenAliceSessionExistsForBob_whenCheckingExistence_itShouldReturnTrue() {
+        val aliceKey = (aliceClient.createInitialPreKeys() as Either.Right).b.lastKey
+        bobClient.createSessionIfNeeded(aliceSessionId, aliceKey)
+
+        bobClient.doesSessionExists(aliceSessionId)
+            .shouldSucceed {
+                it shouldBeEqualTo true
+            }
+    }
+
     @Test
     fun givenBobWantsToTalkToAlice_whenSendingTheFirstMessageAndTheSessionIsAsserted_itShouldBeEncryptedSuccessfully() {
```

Comment:
It should be changed to` whenSendingTheFirstMessageAndTheSessionIsCreated`, right ?

Reply:
Yes, and welcome back!

I'll tackle that in another PR.